src/sourceview.rs

```diff
@@ -139,16 +140,16 @@ impl<'a> Iterator for Lines<'a> {
 /// operations.
 pub struct SourceView<'a> {
     source: Cow<'a, str>,
-    processed_until: RefCell<usize>,
-    lines: RefCell<Vec<(*const u8, usize)>>,
+    processed_until: Arc<Mutex<usize>>,
+    lines: Arc<Mutex<Vec<(*const u8, usize)>>>,
```

Comment:
I don’t think we need any `Arc` here at all, a `Mutex` should suffice.
The first one can also be a `AtomicUsize` instead. I’m unsure why the previous code was using a `RefCell` instead of just a `Cell`.